### PR TITLE
Enable TDX CI

### PR DESCRIPTION
**Pull request by Hsy-Intel**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,8 +70,10 @@ CARGO_OSDK_ARGS += --grub-boot-protocol=$(BOOT_PROTOCOL)
 endif
 
 ifeq ($(ENABLE_KVM), 1)
+ifneq ($(INTEL_TDX), 1)
```
Comment (tatetian) [inline Makefile:73]: Why can't KVM be enabled for TDX?

**Comment by tatetian** [comment]
By the way, can we also test OSDK for TDX in CI?

File: Makefile (hunk 1 above)
Comment (Hsy-Intel) [reply Makefile:73]: Intel TDX currently only supports KVM. This configuration is directly reflected in the QEMU parameters of TDX:
`-accel kvm`

**Comment by junyang-zh** [comment]
> By the way, can we also test OSDK for TDX in CI?

OSDK seems to be a neutral tool agnostic to TDX. It provides the toolings and configurations to allow the user building TDX-compatible OSes. In what aspect OSDK should have tests targeting TDX?

**Comment by Hsy-Intel** [comment]
> > By the way, can we also test OSDK for TDX in CI?
> 
> OSDK seems to be a neutral tool agnostic to TDX. It provides the toolings and configurations to allow the user building TDX-compatible OSes. In what aspect OSDK should have tests targeting TDX?

We will test for booting a "Hello world" OS in TDX cfg.

File: .github/workflows/test_asterinas.yml [hunk 2]
```diff
@@ -86,3 +86,56 @@ jobs:
       - name: General Test (Linux EFI Handover Boot Protocol)
         id: test_linux
         run: make run AUTO_TEST=test ENABLE_KVM=0 BOOT_PROTOCOL=linux-efi-handover64 RELEASE=1
+
+  integration-test-tdx:
+    runs-on: self-hosted
+    timeout-minutes: 30
+    container:
+      image: asterinas/asterinas:0.6.0-tdx
```
[the remark this answers is not in the document]
Comment (Hsy-Intel) [reply .github/workflows/test_asterinas.yml:94]: Will fix in this PR.

File: Makefile (hunk 1 above)
Comment (Hsy-Intel) [reply Makefile:80]: Answer from GPT4:

Both --enable-kvm and -accel kvm are options used with QEMU, a hardware virtualization software, to accelerate virtual machine performance by using hardware virtualization extensions provided by modern CPUs. However, they are used in different versions or command syntaxes of QEMU.

--enable-kvm: This is the older syntax used in QEMU command-line options to enable KVM (Kernel-based Virtual Machine) acceleration. When you pass --enable-kvm to the QEMU command, it tells QEMU to use the KVM interface for virtualization, which allows it to leverage hardware extensions like Intel VT-x or AMD-V to run virtual machines much faster than purely emulated (software-based) virtualization.

-accel kvm: This is the newer syntax introduced in later versions of QEMU to specify the type of accelerator to use. The -accel option is more flexible and allows you to specify different acceleration backends. For example, -accel kvm tells QEMU to use KVM for acceleration, similar to --enable-kvm. It's a more explicit way to define the acceleration backend and is part of a more generalized syntax that can accommodate other types of accelerators as well, such as HAXM on Intel systems or HVF on macOS.

In summary, both options are used to achieve the same goal of enabling KVM acceleration, but -accel kvm is part of a newer, more flexible syntax that allows specifying different acceleration backends, while --enable-kvm is the older, more specific way to enable KVM. If you're using a recent version of QEMU, it's recommended to use the -accel option.

Comment (tatetian) [reply Makefile:80]: So basically, they are the same. We should use one of them, not both. If the QEMU installed in our Docker images support is new enough to support `-accel kvm`, then we should use this recommended argument. If not, we should use `--enable-kvm` for both TDX and non-TDX settings.

File: .github/workflows/test_asterinas.yml [hunk 3]
```diff
@@ -86,3 +86,56 @@ jobs:
       - name: General Test (Linux EFI Handover Boot Protocol)
         id: test_linux
         run: make run AUTO_TEST=test ENABLE_KVM=0 BOOT_PROTOCOL=linux-efi-handover64 RELEASE=1
+
+  integration-test-tdx:
```
Comment (tatetian) [inline .github/workflows/test_asterinas.yml:90]: I think `integration-test` and `integration-test-tdx` should run in parallel to reduce our wait time for the Ci results to come out. Currently, `integration-test` is already the longest running test. If `integration-test-tdx` runs behind it, our CI test will be lengthened significantly.

Comment (Hsy-Intel) [reply .github/workflows/test_asterinas.yml:90]: Currently they are tested in parallel because they are defined in different jobs.

**Comment by Hsy-Intel** [comment]
It looks like TDX CI failed again after 3 attempts...

File: .github/workflows/test_asterinas.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hsy-Intel) [reply .github/workflows/test_asterinas.yml:94]: I find that there are other changes to the OSDK doc, so I'll fix them in the OSDK CI PR.

**Comment by Hsy-Intel** [comment]
> It looks like TDX CI failed again after 3 attempts...

@grief8 is Trying to solve the network problem.
BTW, Github CI can't support scheduled CI for a specific job.

**Comment by tatetian** [comment]
I am afraid that the network connection of the self-hosted CI server is still not stable enough to do CI on every commit. I think we have no choice but to do the TDX CI test on a nightly basis, rather on a per-commit basis.

**Comment by Hsy-Intel** [comment]
> I am afraid that the network connection of the self-hosted CI server is still not stable enough to do CI on every commit. I triggered a benchmark CI manually on 10pm Beijing time. The CI failed due to a timeout: the new Docker image was being downloaded for 1 hour... I think we have no choice but to do the TDX CI test on a nightly basis, rather on a per-commit basis.

Scheduled test added.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!
